### ISSUE TITLE
Exception 

### ISSUE THREAD

**Issue by TravisJoe** [theiviaxx/photoshopConnection#3]
Hello,

I am trying the test script with PS CC 2015 and I receive this when I run it. Is this something you have seen before? Am I handling the message incorrectly?

```shell
File "/Users/travist/SoftwareDev/NoiseBox/pyps/pyps.py", line 214, in run
    self._ids[msg.id] = item
AttributeError: 'Message' object has no attribute 'id'
```

**Comment by theiviaxx** [comment]
I havent had a chance to try it with cc 2015.  They may have changed the protocol slightly.  I'll see what needs to be chnaged

**Comment by TravisJoe** [comment]
Thanks, I appreciate the time. I will look into it more also but I have just started to work with PS connection.

**Comment by theiviaxx** [comment]
so i tested with 2015 and it seems to work fine

* did you set up Remote Connections in photoshop?
* can you paste a sample of the code you were trying to use?

**Comment by TravisJoe** [comment]
Sure no problem. I am using remote connections in Photoshop and I am able to receive the "hello" message. It only has the exception when trying to subscribe to the tool change event.

Something that might be worth noting is I am using the py_des and not the Pycrypto.

The test code
``` python
from screen import GUI
from PSconnect import PSconnect
import time

from pyps import Connection, EventListener

def callback(message, *args):
    print message.command
    print message.content

def callback2(message, *args):
	print message.command
	print message.content
	print args

if __name__ == "__main__":

    psConn = PSconnect()
    # psConn.daemon = True
    psConn.start()

    conn = Connection()
    conn.connect(passwd='NoiseBox')
    conn.send('alert("Hello");', True)

    print conn.send('$.version;', True)

    listener = EventListener(conn)
    listener.start()

    listener.subscribe('toolChanged', callback2, (True, 'xxx'))

	# listener.subscribe('foregroundColorChanged', callback)
	# listener.subscribe('currentDocumentChanged', callback)

    # gui = GUI()

    while True:
	    time.sleep(1.0)
```


The full exception
```shell
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/Users/travist/SoftwareDev/NoiseBox/pyps/pyps.py", line 215, in run
    self._ids[msg.id] = item
AttributeError: 'Message' object has no attribute 'id'

<undefined:>
Traceback (most recent call last):
  File "main.py", line 20, in <module>
    psConn.start()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/multiprocessing/process.py", line 120, in start
    assert self._popen is None, 'cannot start a process twice'
AttributeError: 'PSconnect' object has no attribute '_popen'

***Repl Closed***
```

**Comment by theiviaxx** [comment]
ok so i did find a problem, i changed a constructor signature and things broke.  The EventListener needs to take a passwd arg and it will create its own Connection object.  I will be adding support for both so you can reuse Connection objects, but haven't had time to update it yet.

Try that change and let me know if its still busted

**Comment by TravisJoe** [comment]
Yes thank you, this resolved the issue. One exception was my problem and fixed that. Now I am able to get the tool change events as needed. 

I am not closing this as I am not sure if you want to tie it to the changes to take both arguments.

**Comment by theiviaxx** [comment]
fixed with 85778bf7a2cbe162f42c30696d7b9cc8d869a6ad

**Comment by TravisJoe** [comment]
Thank you sir!

I appreciate the time and hope this was helpful as well.
